src/httpsv_generate.c

```diff
@@ -704,6 +704,22 @@ void HTTPSV_GenerateDemoListing(cluster_t *cluster, oproxy_t *dest)
 	HTTPSV_SendHTMLFooter(cluster, dest);
 }
 
+void HTTPSV_GenerateDemoFilenamesTxt(cluster_t *cluster, oproxy_t *dest)
+{
+	dest->_bufferautoadjustmaxsize_ = 1024 * 1024; // NOTE: this allow 1MB buffer...
```

Comment:
You should always check a pointer's validity before dereferencing it.
If `dest` happens to be NULL when this code is reached, without the extra guard you can segfault the application.

So use
_if (dest != NULL)
{
    dest->_bufferautoadjustmaxsize_ = 1024 * 1024;
}_



I guess there are probably 0 occasions in the sourcecode of qtv where this is done ;), but that doesn't mean we should not add safe code.